resources/benchmark-runner.mjs

```diff
@@ -145,6 +185,31 @@ class PageElement {
         const event = new contentWindow.MouseEvent(type, eventOptions);
         this.#node.dispatchEvent(event);
     }
+
+    _getParent(path) {
+        const parent = path.reduce((root, selector) => {
+            const node = root.querySelector(selector);
+            return node.shadowRoot ?? node;
+        }, this.#node.shadowRoot ?? this.#node);
+
+        return parent;
+    }
```

Comment:
nit: Could you move this to a separate shared helper function that also takes a `lookupStartNode` or so?

Reply:
done